pkg/asset/agent/manifests/infraenv.go

```diff
@@ -36,49 +39,49 @@ func (*InfraEnv) Name() string {
 func (*InfraEnv) Dependencies() []asset.Asset {
 	return []asset.Asset{
 		&agent.OptionalInstallConfig{},
-		// &AgentPullSecret{},
+		&NMStateConfig{},
 	}
 }
 
 // Generate generates the InfraEnv manifest.
 func (i *InfraEnv) Generate(dependencies asset.Parents) error {
 
-	// installConfig := &agent.OptionalInstallConfig{}
-	// agentPullSecret := &AgentPullSecret{}
-	// dependencies.Get(installConfig, agentPullSecret)
-
-	// infraEnv := &aiv1beta1.InfraEnv{
-	// 	ObjectMeta: v1.ObjectMeta{
-	// 		Name:      "infraEnv",
-	// 		Namespace: installConfig.Config.Namespace,
-	// 	},
-	// 	Spec: aiv1beta1.InfraEnvSpec{
-	// 		ClusterRef: &aiv1beta1.ClusterReference{
-	// 			Name:      installConfig.Config.ObjectMeta.Name,
-	// 			Namespace: installConfig.Config.ObjectMeta.Namespace,
-	// 		},
-	// 		SSHAuthorizedKey: installConfig.Config.SSHKey,
-	// 		PullSecretRef: &corev1.LocalObjectReference{
-	// 			Name: agentPullSecret.ResourceName(),
-	// 		},
-	// 		// NMStateConfigLabelSelector: v1.LabelSelector{
-	// 		// 	MatchLabels: map[string]string{
-	// 		// 		// fetch from NMStateConfig
-	// 		// 	},
-	// 		// },
-	// 	},
-	// }
-	// i.Config = infraEnv
-
-	// infraEnvData, err := yaml.Marshal(infraEnv)
-	// if err != nil {
-	// 	return errors.Wrap(err, "failed to marshal agent installer infraEnv")
-	// }
-
-	// i.File = &asset.File{
-	// 	Filename: infraEnvFilename,
-	// 	Data:     infraEnvData,
-	// }
+	installConfig := &agent.OptionalInstallConfig{}
+	nmStateConfig := &NMStateConfig{}
+	dependencies.Get(installConfig, nmStateConfig)
+
+	if installConfig.Config != nil && nmStateConfig != nil {
+		infraEnv := &aiv1beta1.InfraEnv{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "infraEnv",
+				Namespace: installConfig.Config.Namespace,
+			},
+			Spec: aiv1beta1.InfraEnvSpec{
+				ClusterRef: &aiv1beta1.ClusterReference{
+					Name:      installConfig.Config.ObjectMeta.Name,
+					Namespace: installConfig.Config.ObjectMeta.Namespace,
+				},
+				SSHAuthorizedKey: strings.Trim(installConfig.Config.SSHKey, "|\n\t"),
+				PullSecretRef: &corev1.LocalObjectReference{
+					Name: "pull-secret",
```

Comment:
Should use `agentPullSecretName` instead of a literal.

Reply:
We will go with `<clustername>-pull-secret` as discussed in slack with @zaneb